src/main/host/syscall/handler/mod.rs

```diff
@@ -319,7 +321,25 @@ impl SyscallHandler {
             // UNSUPPORTED SYSCALL
             //
             _ => {
-                log::warn!("Detected unsupported syscall {} ({}) called from thread {} in process {} on host {}",
+                // only show a warning the first time we encounter this unsupported syscall
+                static WARNED_SET: Mutex<Option<HashSet<SyscallNum>>> = Mutex::new(None);
+
+                // `insert()` returns `false` if the syscall num was already in the set
+                let has_already_warned = !WARNED_SET
+                    .lock()
+                    .unwrap()
+                    .get_or_insert_with(HashSet::new)
+                    .insert(syscall);
```

Comment:
I assume this is global across all shadow worker threads? I guess maybe we don't care because we expect that unsupported syscalls won't happen super frequently so lock contention will not be of practical concern?

Reply:
Yeah this is global across all threads, and yeah I'm assuming that it will be uncommon for two threads to be handing an unsupported syscall at the same time. The lock is only held for the hash set lookup (and possibly insertion), so assuming that unsupported syscalls aren't called very often, and a hash set lookup is fast, I don't expect there to be much if any contention. I could change this to an `RwLock`, but it would make the code a bit more complicated (first try acquiring the read lock, then possibly try acquiring the write lock).

Reply:
OK, thanks for the explanation! No need to change the code I think.

Reply:
I changed this to an `RwLock` anyways, I'm not really sure if an `RwLock` is slower in the general case, but at least we won't have to worry about contention.